status-im/status-desktop
# chore: bump status-go with filter improvements

### What does the PR do

Improved lightClient functionality especially fixing issues around Waku-Filter protocol usage and optimizing how filter subscriptions are managed and Filter ping occurs with service node. 
Still some optimizations and additional work is pending wrt light client reliability, but this PR brings Filter to a hopefully decent stage(without any major issues) before proceeding with further changes.

Note that if you are subscribed to status community, there would be a lot of filter subscriptions. This still needs optimization which is pending. Anyways, number of subscriptions would help test nwaku node as well to handle soo may :)

**Note that to reduce issues with lightClient(by not connecting to other desktop instances), this build only uses fleet nodes and does not do any discovery via peerExchange.**

Relevant status-go PR https://github.com/status-im/status-go/pull/4665

### Affected areas
LightMode functionality

### How to test
Note to enable debug logs while testing.
- Enable lightMode is desktop from Advanced Settings
- Login normally, and make sure you got some peers connected in the node management tab
- Run tests by being part of community , 1:1 chats, private group chats etc
- Disconnect your machine from the internet, wait until the peer count goes to 0. You should see following log in `geth.log`

```
going offline, removing all filter subscriptions
```

- Connect to the internet, you should see the following log line in `geth.log`:
- You should reconnect to some peers.
- Test reception of messages on 1:1 chat, groups and community
- Leave it running, let your machine go to sleep and wake-up etc and ensure no issues in send/receive messages

**In case of any issue please report along with debug logs via DM or as a comment in this PR itself.**
